hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaMessageReceiverFactory.java

```diff
@@ -98,7 +101,11 @@ public MessageReceiver createMessageReceiver(Topic topic,
         props.put(SESSION_TIMEOUT_MS_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_SESSION_TIMEOUT_MS_CONFIG));
         props.put(HEARTBEAT_INTERVAL_MS_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG));
         props.put(METADATA_MAX_AGE_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_METADATA_MAX_AGE_CONFIG));
-        props.put(MAX_PARTITION_FETCH_BYTES_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_MAX_PARTITION_FETCH_BYTES_CONFIG));
+
+        SubscriptionFetchingConfiguration fetchingConfiguration = subscription.getFetchingConfiguration();
+        int globalMaxBytes = configs.getIntProperty(Configs.KAFKA_CONSUMER_MAX_PARTITION_FETCH_BYTES_CONFIG);
+        props.put(MAX_PARTITION_FETCH_BYTES_CONFIG, min(fetchingConfiguration.getMaxPartitionFetchBytes(), globalMaxBytes));
```

Comment:
This option is intended to be a higher bound, if we also implement a lower bound then we can enforce minimum buffer size. What you're suggesting would work if we store in zk some sort of pointer to global value and not the value itself. Then we can change that dynamically like that.